lib/Widget/Render/WidgetHtmlRenderer.php

```diff
@@ -468,6 +468,21 @@ private function render(
                 $widget->isValid = 0;
             }
 
+            $moduleLanguage = null;
+            $translator = null;
+
+            // Check if a language property is defined against the module
+            // Note: We are using the language defined against the module and not from the module template
+            foreach ($module->properties as $property) {
+                if ($property->id === 'lang' && !empty($property->value)) {
```

Comment:
I still think this should be based on the variant `languageSelector` rather than the property ID, in case someone uses the language selector on a property with a different id.

Reply:
This makes sense.